tfe/data_source_oauth_client.go

```diff
@@ -0,0 +1,65 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceTFEOAuthClient() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceTFEOAuthClientRead,
+		Schema: map[string]*schema.Schema{
+			"oauth_client_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"api_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"http_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"oauth_token_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ssh_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
```

Comment:
i've confirmed that the rsa_public_key is only every populated for bitbucket server connections and isn't something that would be useful so it can just be removed.